pymeasure/instruments/validators.py

```diff
@@ -40,6 +40,30 @@ def strict_range(value, values):
         ))
 
 
+def strict_discrete_range(value, values, step):
+    """ Provides a validator function that returns the value
+    if its value is less than the maximum and greater than the
+    minimum of the range and is a multiple of step.
+    Otherwise it raises a ValueError.
+
+    :param value: A value to test
+    :param values: A range of values (range, list, etc.)
+    :param step: Minimum stepsize (resolution limit)
+    :raises: ValueError if the value is out of the range
+    """
+    if (min(values) <= value <= max(values)):
+        if (value/step).is_integer():
```

Comment:
I think it is preferable to use the modulus to determine if it is a multiple: `value % step == 0`

Reply:
This was my first intention as well, but I recognized that the modulus does not work very reliably with decimals:
`0.1 % 0.02 == 0` returns `False`
whereas the proposed
`(0.1/0.02).is_integer()` gives `True`
Do you have any idea how to fix this? It is probably due to some accuracy limit in number representation, since the modulus gives a very small (E-18) close to zero value.
The chosen variant is not as elegant as using the modulus but it works :D

Reply:
I thinks this is mainly down to floating point arithmetic. `math.isclose()` with appropriate tolerances is probably the safest/cleanest way to handle this.

Reply:
thanks for your comment! Upon your suggestion I tried `math.isclose()` and found a counterexample: `math.isclose(0,0.6%0.1,rel_tol=1e-9,abs_tol=1e-2)` evaluates to `False` despite the "huge" absolute tolerance. Are there any obejctions against the `Decimal` type solution? Computing time? As far as I understand this datatype is intended especially for such scenarios.

Reply:
I think `Decimal` is a good solution for this use case. Thanks!